### ISSUE TITLE
Keyboard or focusNode

### ISSUE THREAD

**Issue by klaszlo8207** [jonataslaw/get#22]
Is there any way to use nextFocus without a context? Because when my TextField in a widget that is opened via Get my keyboard is going to be crazy from this line:

onSubmitted: (_) => FocusScope.of(context).nextFocus(), // move focus to next
        ),

**Comment by jonataslaw** [comment]
> Is there any way to use nextFocus without a context? Because when my TextField in a widget that is opened via Get my keyboard is going to be crazy from this line:
> 
> onSubmitted: (_) => FocusScope.of(context).nextFocus(), // move focus to next
> ),

You can retrieve the context of the last builder widget using Get.key.currentWidget from anywhere in your code.

onSubmitted: (_) => FocusScope.of(Get.key.currentContext).nextFocus(), // move focus to next
),

**Comment by klaszlo8207** [comment]
Sorry but this is not working at all:

next Focus never call when I use with **Get.key.currentContext**

Also when I use this:

```
TextField(
        textInputAction: TextInputAction.next,
        controller: controller,
 onSubmitted: (s) {
          FocusScope.of(context).nextFocus();
          //FocusScope.of(Get.key.currentContext).nextFocus();
        },
```

then the probeleme is:

First time iot is working like a charm, But after I write something in the field and submit called setState or changeNotifier, the second time when click in the TextField my keyboard just pop up and immediatelly pop down (show and hide)

Also I have to generate on the fly a lot of TextField and controllers for that and perhaps listeners to that...

Can you help me how to fix this?

Basically what I want:

Row1 [Textfield][Textfield][Textfield]
Row2 [Textfield][Textfield][Textfield]
Row3 [Button to add new Row]

**ADD ->**

Row1 [Textfield][Textfield][Textfield]
Row2 [Textfield][Textfield][Textfield]
Row3 [Textfield][Textfield][Textfield]
Row4 [Button to add new Row]

**DEL ->**

Row1 [Textfield][Textfield][Textfield]
Row2 [Textfield][Textfield][Textfield]
Row3 [Button to add new Row]

Also I have to save those values in somehow and refresh the UI, via setState or **changeNotifier**


**I also find this BUG in the flutter app:**

https://github.com/flutter/flutter/issues/52342#issuecomment-599456423

**Comment by klaszlo8207** [comment]
ere is my textfield code

```
TextField(
        textInputAction: TextInputAction.next,
        controller: controller,
        maxLength: 3,
        maxLines: 1,
        /*
        onChanged: (s) {
          if (onChanged != null) onChanged(s);
        },*/
        /*
        onSubmitted: (s) {
          //FocusScope.of(Get.key.currentContext).nextFocus();
          FocusScope.of(context).nextFocus(); //EZ BASSZA EL?!!!!!!  // minden of(context) !!!
          //if (onSubmitted != null) onSubmitted(s);
        },*/
        textAlign: TextAlign.center,
        enableInteractiveSelection: !disableCopyPaste,
        keyboardType: TextInputType.number,
        inputFormatters: <TextInputFormatter>[WhitelistingTextInputFormatter.digitsOnly],
        cursorColor: Colors.black,
        decoration: InputDecoration(
          counterText: "",
          border: InputBorder.none,
        ),
        style: textStyleNormal,
      ))

```
and my listeners generating in realtime after build something like this:

```
 final textWeightControllerListener = () {
          final newText = textWeightController.text;
          textWeightController.value = textWeightController.value.copyWith(
            text: newText,
            selection: TextSelection(baseOffset: newText.length, extentOffset: newText.length),
            composing: TextRange.empty,
          );
          _updateSetFor(exercises[i], j, weight: textWeightController.text);
        };
        textWeightController.addListener(textWeightControllerListener);
```

**Comment by jonataslaw** [comment]
You are probably swapping some StatefulWidget for StatelessWidget, or inserting controllers inside build methods instead of inserting them at the beginning of a Stateful class.

**Comment by klaszlo8207** [comment]
So basically I have to create all the TextEditingControllers in initState instead of build? That will work?

**Comment by jonataslaw** [comment]
> So basically I have to create all the TextEditingControllers in initState instead of build? That will work?

Just like I told you in that comment:
https://github.com/jonataslaw/get/issues/20#issuecomment-597416705

Controllers can NEVER be inside builders, no matter if it’s the main build (context) widget, or inside a StreamBuilder, FutureBuilder, BottomSheet, because at each setState a new controller will be created.

You can never do this:
```dart
class Example extends StatelessWidget {
  const Example({Key key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final _editText = TextEditingController(); // NEVER DO THAT. Controllers should never be within builder methods.
    return TextField(
      controller: _editText,
    );
  }
}

```
You should ALWAYS do this:

```dart
class Example extends StatefulWidget { 
  // If you have any controller it is mandatory to use StatefulWidget.
  Example({Key key}) : super(key: key);

  @override
  _ExampleState createState() => _ExampleState();
}

class _ExampleState extends State<Example> {
 final TextEditingController _editText = TextEditingController(); // HERE

  @override
  Widget build(BuildContext context) {
    return TextField(
      controller: _editText,
    );
  }
}

```
You can also create the controller in the mentioned place, and instantiate it in your initState, so it will only be called once, however, for code security reasons, I recommend doing it as above.

Note that this is not related to this library, and as much as I want to help you, it can mess up people who are looking for something in issues.
If you want to ask questions about the framework not related to this library, you can ask me on the messenger
https://www.facebook.com/jonny.borges.982

Reserve this field only for issues directly related to Get.
